### PR TITLE
[REG-1989] Add UniRx tests to RGUnityBots + Handle sending pointer events for touches

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Handles touch 'pointer' events in our legacy input code
  - Unit tests added for this
- Adds UniRx extensions to our RGUnityBots project to enable testing of pointer events
- Handles touch 'pointer' recording in our system.  Plays back touch events as left mouse button clicks as a preliminary solution.
  - Tested in bossroom by using the InputDebugger to simulate touch input from mouse and then recording/playing back and confirming correct function
---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: src/gg.regression.unity.bots/Runtime/Scripts/RGLegacyInputUtility/RGLegacyInputWrapper.cs [hunk 1]
```diff
@@ -431,6 +433,17 @@ private static void SendMouseEventMessage(GameObject gameObject, string methodNa
             }
         }
 
+        private static void SendPointerEvent(GameObject gameObject, string methodName)
```
Comment (RG-nAmKcAz) [inline src/gg.regression.unity.bots/Runtime/Scripts/RGLegacyInputUtility/RGLegacyInputWrapper.cs:436]: This is the code change that actually makes pointer events work in our legacy input code

File: src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/MouseInputActionObserver.cs [hunk 2]
```diff
@@ -112,25 +112,45 @@ public void ObserveMouse(IEnumerable<ObjectStatus> statefulObjects)
 
         public MouseInputActionData GetCurrentMouseState(Vector2? position = null)
         {
-            var mouse = Mouse.current;
-            if (mouse != null)
+            var pointer = Pointer.current;
```
Comment (RG-nAmKcAz) [inline src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/MouseInputActionObserver.cs:115]: This is our preliminary support for touch/pen pointer events in the current InputSystem during recording/playback.

File: src/RGUnityBots/Assets/Scripts/FPSCap.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+using UnityEngine;
+
+public class FPSCap : MonoBehaviour
+{
+    public int targetFrameRate = 30;
+
+    private void Start()
+    {
+        QualitySettings.vSyncCount = 0;
```
Comment (RG-nAmKcAz) [inline src/RGUnityBots/Assets/Scripts/FPSCap.cs:9]: This is here to limit the amount of laptop fan noise while testing this input project as we really don't need it running at max fps/cpu

File: src/RGUnityBots/Assets/Scripts/LegacyInputTests/LegacyPointerHandler.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+            gameObject.OnPointerDownAsObservable()
+                .SelectMany(_ => this.gameObject.UpdateAsObservable())
+                .TakeUntil(this.gameObject.OnPointerUpAsObservable())
+                .Select(_ => Input.mousePosition)
+                .RepeatUntilDestroy(this) // safety way
+                .Subscribe(
+                    x =>
+                    {
+                        Debug.Log($"{gameObject.name} OnPointerDownAsObservable()");
```
Comment (RG-nAmKcAz) [inline src/RGUnityBots/Assets/Scripts/LegacyInputTests/LegacyPointerHandler.cs:59]: Adds the logging we use in unit tests to confirm that UniRx and touch inputs work.

File: src/RGUnityBots/Assets/Scripts/LegacyInputTests/LegacyMouseHandler.cs [hunk 5]
```diff
@@ -1,8 +1,34 @@
...
+    {
+        gameObject.OnMouseDownAsObservable()
+            .SelectMany(_ => this.gameObject.UpdateAsObservable())
+            .TakeUntil(this.gameObject.OnMouseUpAsObservable())
+            .Select(_ => Input.mousePosition)
+            .RepeatUntilDestroy(this) // safety way
+            .Subscribe(
+                x =>
+                {
+                    Debug.Log($"{gameObject.name} OnMouseDownAsObservable()");
```
Comment (RG-nAmKcAz) [inline src/RGUnityBots/Assets/Scripts/LegacyInputTests/LegacyMouseHandler.cs:17]: Adds the logging we use in unit tests to confirm that UniRx and mouse inputs work.

**Review by addisonbgross — APPROVED** [review]
LGTM. Just 1 small non-blocking question

File: src/RGUnityBots/Assets/Plugins/UniRx/Examples/Sample01_ObservableWWW.cs [hunk 6]
```diff
@@ -0,0 +1,90 @@
+﻿#if !(UNITY_METRO || UNITY_WP8)
```
[the remark this answers is not in the document]
Comment (RG-nAmKcAz) [reply src/RGUnityBots/Assets/Plugins/UniRx/Examples/Sample01_ObservableWWW.cs]: i removed these as noted, but this shouldn't impact the sdk download size as customers see it, only us as developers cloning the entire repo

**Review by batu — APPROVED** [review]
Thank you for pointing out relevant changes

File: src/gg.regression.unity.bots/Tests/Scripts/LegacyInput/RGLegacyInputSimulationTests.cs [hunk 7]
```diff
@@ -17,6 +20,24 @@ public class RGLegacyInputSimulationTests
         private void ResetState()
         {
             RGLegacyInputWrapper.StopSimulation();
+
+
+            TouchSimulation.Enable();
```
Comment (vontell) [inline src/gg.regression.unity.bots/Tests/Scripts/LegacyInput/RGLegacyInputSimulationTests.cs:25]: Thank you for adding the tests!

**Review by vontell — APPROVED** [review]
